common/app/common/dfp/DfpData.scala

```diff
@@ -148,13 +149,13 @@ case class GuTargeting(adUnitsIncluded: Seq[GuAdUnit],
                        geoTargetsExcluded: Seq[GeoTarget],
                        customTargetSets: Seq[CustomTargetSet]) {
 
-  val keyValues: Seq[String] = {
-    for {
-      targetSet <- customTargetSets
-      target <- targetSet.targets if target.isKeywordTag
-      targetValue <- target.values
-    } yield targetValue
-  }
+  // This is local pure helper function:
+  private val tagValues = ( ( customTargetSets: Seq[CustomTargetSet],
+                              isGoodTag: CustomTarget => Boolean )
```

Comment:
Maybe a better name for `isGoodTag` is `tagFilter` because this function can literally filter the tags in a non prescribed way.